backend/users/views.py

```diff
@@ -0,0 +1,29 @@
+import json
+from base64 import urlsafe_b64decode
+
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from social_django.models import UserSocialAuth
+
+
+class UserInfoView(APIView):
+    def get(self, request: Request, *args, **kwargs) -> Response:
+        try:
+            user_info = UserSocialAuth.objects.get(user=request.user)
+        except UserSocialAuth.DoesNotExist as error:
+            from hitas.exceptions import HitasModelNotFound
+
+            raise HitasModelNotFound(UserSocialAuth) from error
+
+        raw_token = user_info.extra_data["id_token"].split(".")[1]
+        padded_token = raw_token + "=" * divmod(len(raw_token), 4)[1]
+        token_payload = json.loads(urlsafe_b64decode(padded_token))
+
+        data = {
+            "first_name": token_payload.get("given_name"),
+            "last_name": token_payload.get("family_name"),
+            "email": token_payload.get("email"),
+        }
+        return Response(data=data, status=status.HTTP_200_OK)
```

Comment:
I would add a fallbacks here.
- If the `UserSocialAuth` object does not have a first or last name, we can get them from our own `User` model.
- Additionally if we want to keep using Django users when testing, we should fallback to using them if `UserSocialAuth.objects.get(user=request.user)` is not found 😉

Reply:
A fallback to `User` model first name, last name, and email make sense, I'll implement that.

Currently, I haven't changed how regular tests create the authenticated user (`UserSocialAuth` model is not created in `hitas.tests.conftest.api_client`). `DEFAULT_AUTHENTICATION_CLASSES` still has both DEV and Helsinki profile tokens on, so tests will continue to work normally until that is changed.